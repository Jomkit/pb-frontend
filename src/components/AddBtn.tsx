const AddBtn = ({handleClick, btnName}: {handleClick: () => void, btnName: string}) => {
    
  return (
    <div className='text-start'>
        <button className='btn bg-green-400 border-solid border-2 border-black' onClick={handleClick}>New {btnName}</button>
    </div>
  )
}

export default AddBtn