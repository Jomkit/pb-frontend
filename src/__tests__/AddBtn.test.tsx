import { render } from "@testing-library/react"
import AddBtn from "../components/AddBtn"

describe("AddBtn", function(){
    const testAdd = vi.fn();

    it('should render without crashing', () => { 
        render(<AddBtn handleClick={testAdd} btnName="testAddBtn" />);
    })

    it('should match snapshot', () => { 
        const { asFragment } = render(<AddBtn handleClick={testAdd} btnName="testAddBtn" />)
        expect(asFragment()).toMatchSnapshot();
    })
})