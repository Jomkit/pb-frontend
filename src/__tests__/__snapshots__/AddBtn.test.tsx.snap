// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddBtn > should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="text-start"
  >
    <button
      class="btn bg-green-400 border-solid border-2 border-black"
    >
      New testAddBtn
    </button>
  </div>
</DocumentFragment>
`;
